tipg/filter/filters.py

```diff
@@ -58,7 +58,11 @@ class Operator:
         "not_in": lambda f, a: ~f == any(a),
         "any": lambda f, a: f.any(a),
         "not_any": lambda f, a: f.not_(f.any(a)),
-        "INTERSECTS": lambda f, a: Func("st_intersects", f, a),
+        "INTERSECTS": lambda f, a: Func(
+            "st_intersects",
+            f,
+            Func("st_setsrid", cast(a, "geometry"), Func("st_srid", f)),
```

Comment:
st_setsrid does not actually transform the geometry to the new srid, to actually change the geometry srid you need to use st_transform(geometry, srid)